### ISSUE TITLE
Valid JSON Example Triggers Swagger Spec Failure

### ISSUE THREAD

**Issue by cjslep** [go-openapi/validate#43]
Brief backstory: Was investigating an issue with go-swagger failing to generate client code for Eve Online's latest swagger spec.

Long story short, this file will fail validation:

```
{
    "swagger": "2.0",
    "info": {
        "title": "Test",
        "description": "Tests issue with examples and nested objects with items properties",
        "version": "1.0"
    },
    "host": "FooBar",
    "basePath": "/latest",
    "schemes": [
        "https"
    ],
    "produces": [
        "application/json"
    ],
    "paths": {
        "/foo/": {
            "get": {
                "description": "Tests issue with nested items object properties and examples",
                "summary": "Tests issue with nested items object properties and examples",
                "responses": {
                    "200": {
                        "description": "Example should match spec",
                        "examples": {
                            "application/json": [
                                {
                                    "items": [
                                        {
                                            "id": 1
                                        }
                                    ]
                                }
                            ]
                        },
                        "schema": {
                            "type": "array",
                            "items": {
                                "type": "object",
                                "properties": {
                                    "items": {
                                        "type": "array",
                                        "maxItems": 255,
                                        "items": {
                                            "type": "object",
                                            "properties": {
                                                "id": {
                                                    "type": "integer",
                                                    "format": "int32"
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```

The test I added to `spec_test.go` was similar to the other issue tests, if the above is put into `fixtures/bugs/new/swagger.json`:

```
func TestNewIssue(t *testing.T) {
	fp := filepath.Join("fixtures", "bugs", "new", "swagger.json")
	jstext, _ := ioutil.ReadFile(fp)

	// as json schema
	var sch spec.Schema
	if assert.NoError(t, json.Unmarshal(jstext, &sch)) {
		validator := NewSchemaValidator(spec.MustLoadSwagger20Schema(), nil, "", strfmt.Default)
		res := validator.Validate(&sch)
		assert.True(t, res.IsValid())
	}

	// as swagger spec
	doc, err := loads.Spec(fp)
	if assert.NoError(t, err) {
		validator := NewSpecValidator(doc.Schema(), strfmt.Default)
		res, _ := validator.Validate(doc)
		assert.True(t, res.IsValid())
	}

}
```

Note that the first assert passes, the second fails:

```
--- FAIL: TestNewIssue (8.64s)
        Error Trace:    spec_test.go:79
	Error:      	Should be true
FAIL
exit status 1
```

Note that deleting the `examples` object will cause the spec parsing to be successful.

Note that I also tried 'peeling' back the outermost layer of the `schema`, by only making it an object with an items property that is an array. The example also still fails in this case, when adjusted in the same manner.

**Comment by fredbi** [comment]
Tested your spec in my development branch for which examples are validated.

I still got the issue, which is due to the property "items".
If I replace "items" by "Items" or "item" it is valid.

The internal schema validator is misled by your choice of a property which is also a json-schema keyword.

**Comment by casualjim** [comment]
this is a bug on our end, the reserved words are positional in a spec. So items where it occurs in the path has to be valid.
@kreamyx was looking into this because it's part of a series of bug fixes around ref resolution that made this appear.

**Comment by fredbi** [comment]
@casualjim if I summarize our discussion, this is problem in spec handling of properties, not in the package "validate" per se.

It so happens that the validation of examples in responses scans the schema for the items property which is not properly validated, because the property "items" is mismatched with the schema keyword "items" at spec parsing time.

**Comment by cjslep** [comment]
Thanks for the discussion. My apologies for not filing it against the correct repository -- this was the easiest entry point into showing a reproducible issue.

For clarity: I am not an employee of CCP Games. So I have no control over how Eve Online's spec is defined.

I greatly appreciate you all picking up the baton from where I left it off.

**Comment by fredbi** [comment]
The validation part had to take the hit, since it is the one reporting about issues.
I have to admit that in your case, the reported error message was quite terse.
The side effect of adding examples is indeed hard to follow.
As @casualjim pointed out, this is deeper in the internal structure representing the spec.

**Comment by fredbi** [comment]
I am revising my opinion above: this bug is indeed in this package, not go-openapi/spec. 
The special, but wrong, handling of "items" is also the cause of #83
